src/java.base/share/classes/java/lang/invoke/VarHandle.java

```diff
@@ -2067,24 +2063,29 @@ public final MethodType accessModeType(AccessMode accessMode) {
     }
 
     @ForceInline
-    final void checkExactAccessMode(VarHandle.AccessDescriptor ad) {
+    boolean checkExactAccessMode(VarHandle.AccessDescriptor ad) {
         if (exact && accessModeType(ad.type) != ad.symbolicMethodTypeExact) {
             throwWrongMethodTypeException(ad);
         }
+        // return true unless overridden in an IndirectVarHandle
+        return true;
     }
 
     @DontInline
-    private final void throwWrongMethodTypeException(VarHandle.AccessDescriptor ad) {
+    final void throwWrongMethodTypeException(VarHandle.AccessDescriptor ad) {
```

Comment:
This can actually be `private` and `static`:
```suggestion
    private static final void throwWrongMethodTypeException(VarHandle.AccessDescriptor ad) {
```

Reply:
`private` yes. `static` would require to pass the `MethodType` and since this is an exceptional slow-path it doesn't make sense to keep the complexity contained inside the `throw...` method.